src/main/java/com/chainstep/atlas/example/CustomPolicy.java

```diff
@@ -5,27 +5,142 @@
 import id.walt.credentials.w3c.VerifiableCredential;
 import lombok.extern.slf4j.Slf4j;
 import org.jetbrains.annotations.NotNull;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper; 
+import com.fasterxml.jackson.databind.ObjectWriter;
+
+
 @Slf4j
 public class CustomPolicy extends SimpleVerificationPolicy {
+    // Constructor - initializes CustomPolicy
     public CustomPolicy() {
     }
 
-    public boolean getApplyToVC() {
-      return false;
-    }
- 
-    public boolean getApplyToVP() {
-      return true;
-    }
-    
+    //TODO: Please change these Values to define which role of which company is allowed to access yor service.
+    private String legalRegistrationNumString = "DummyLRN";
+    private String roleToGrantAccessString = "DummyRole";
+
+    // Method to return a description of the policy
     @NotNull
     public String getDescription() {
-        return "bla bla bla";
+        return "CustomPolicy checks your defined policies.";
     }
 
+    // Returns false if this policy does not apply to Verifiable Credentials (VCs)
+    public boolean getApplyToVC() {
+        return false;
+     }
+  
+    // Returns true if this policy applies to Verifiable Presentations (VPs)
+    public boolean getApplyToVP() {
+        return true;
+     }
+
+    // The main verification method, checks the VC against custom policy
     @NotNull
     protected VerificationPolicyResult doVerify(@NotNull VerifiableCredential vc) {
-        log.info("CustomPolicy");
-        return VerificationPolicyResult.Companion.success();
+        log.info("CustomPolicy is called");
+
+        try {
+            // Extract the "verifiableCredential" property and convert it to JSON string
+            Object object = vc.getProperties().get("verifiableCredential");
+            ObjectWriter ow = new ObjectMapper().writer().withDefaultPrettyPrinter();
+            String jsonString = ow.writeValueAsString(object);
+            //log.info("returning JSON String: " + jsonString);                      
+
+            ObjectMapper objectMapper = new ObjectMapper();
+            
+            // Parse the JSON string into an array of JsonNodes
+            JsonNode rootNode = objectMapper.readTree(jsonString);
+            
+            // Access the first element in the array (the credential itself)
+            JsonNode credentialNode = rootNode.get(0);
+
+            // Check if the credential is signed by AtlasXAuth
+            if(isSignedByAtlasXAuth(credentialNode)){
+                log.info("isSignedByAtlasXAuth true");   
+            }
+            else{
+                log.info("isSignedByAtlasXAuth false! NO access!");
+                return VerificationPolicyResult.Companion.failure();  // Fail if not signed by AtlasXAuth
+            }
```

Comment:
I believe we don't really need to log "isSignedByAtlasXAuth true" my suggestion would be use this snippet instead for less lines of code:

```java
if (!isSignedByAtlasXAuth(credentialNode)) {
   log.info("isSignedByAtlasXAuth false! NO access!");
   return VerificationPolicyResult.Companion.failure(); 
}
```